engine/src/main/java/org/terasology/input/InputSystem.java

```diff
@@ -420,8 +428,39 @@ private void processControllerInput(float delta) {
         }
     }
 
+
+    /**
+     * Simulated key strokes: To simulate input from a keyboard, we simply have to extract the Input associated to the action
+     * and this function adds it to the keyboard's input queue.
+     * @param key The key to be simulated.
+     */
+    public void simulateSingleKeyStroke(Input key) {
+        /* TODO: Perhaps there is a better way to extract the character.
+            All the simulate functions extract keyChar by getting the first character from it's display string.
+            While it works for normal character buttons, might not work for special buttons if required later.
+        */
+        char keyChar = key.getDisplayName().charAt(0);
```

Comment:
So I can't extend the KeyboardAction class because it's a final class and I don't want to risk breaking anything by making it non static. Instead I decided to make a constructor that doesn't take inputChar. 

> public KeyboardAction(Input input, ButtonState state) {
>         this.input = input;
>         this.state = state;
>     }

But then, inputChar is final too! Do I just give any random character? I just don't know if this inputChar value is required anywhere later. 
As of now, I don't see it getting used anywhere. It's added in LwjglKeyboardDevice, but like not really used anywhere else...